rust/rsc/postgres-tests.wake

```diff
@@ -86,59 +77,47 @@ export target testPostgres Unit =
 
             Pass (dirname postgresPath)
 
-    require Pass cargoPath = findCargoPath Unit
-
-    # This is normally unsafe but we're just using it for
-    # a warning, you should never depend on the result here
-    require _, _ = files "{@here}/vendor" `.*`
-    else failWithError "You forgot to run `cargo vendor --locked` inside {@here}"
+    require Pass testBinary = buildRSCTest Unit
 
     # This script is piped in via stdin so that
     # as shells are opened one after the other,
     # the successive commands are run as intended
     def script =
         """
         set -ex
-        trap 'exit $(cat /tmp/status)' 0
         unshare -U
+        set -ex
         export HOME=/root
         mkdir %{dataDir}
-        echo $? > /tmp/status
         %{postgresDir}/initdb -D %{dataDir}
-        echo $? > /tmp/status
         %{postgresDir}/postgres -D %{dataDir} &
-        trap "kill $!" 0
+        trap "kill $! && sleep 2s" EXIT
         sleep 2s
         %{postgresDir}/createuser -h localhost root
-        echo $? > /tmp/status
         %{postgresDir}/createdb -h localhost shim
-        echo $? > /tmp/status
-        cd %{@here}
-        %{cargoPath} test
-        echo $? > /tmp/status
-        exit $(cat /tmp/status) # exit the unshare shell
+        %{testBinary.getPathName}
         """
 
     # Jobs require that we write stdin out to a file in order to
     # pipe it in
-
     require Pass scriptPath = write ".build/postgres-cargo.sh" script
 
-    def _ = println "Cargo Path: {cargoPath}"
-    def pgJson = postgresSpecJson scriptPath.getPathName cargoPath
+    def pgJson = postgresSpecJson scriptPath.getPathName testBinary.getPathName
 
     require Pass pgJsonPath = write ".build/postgres-cmd-spec.json" (prettyJSON pgJson)
 
-    def emitVendorWarning status =
+    def inspectStatus status =
         require (Exited 0) = status
-        else failWithError "cargo tests failed: Consider running `cargo vendor --locked` in {@here}"
+        else failWithError "cargo tests failed"
 
         Pass status
 
-    makeExecPlan ("{wakePath}/wakebox", "-p", pgJsonPath.getPathName, Nil) Nil
+    makeExecPlan
+    ("{wakePath}/wakebox", "-p", pgJsonPath.getPathName, Nil)
+    (testBinary, scriptPath, pgJsonPath, Nil)
     | editPlanEnvironment (addEnvironmentPath "/usr/sbin")
-    | setPlanPersistence ReRun
```

Comment:
Does one of the input files always change to make the `ReRun` no longer necessary?  I'd expect we'd always want to rerun tests.

Reply:
yeah, two bits of info that matter

1) This isn't a test, its a test runner
2) previously it didn't list any visible files, I've updated it to consume all the files that could invalidate the test (including the test binary itself). Now if any of them change this job will rerun